wxLua/modules/wxlua/wxlstate.cpp

```diff
@@ -2041,11 +2041,10 @@ const char* wxLuaState::lua_SetUpvalue(int funcindex, int n)
 int wxLuaState::lua_SetHook(lua_Hook func, int mask, int count)
 {
     wxCHECK_MSG(Ok(), 0, wxT("Invalid wxLuaState"));
-    // lua_sethook is void in 5.3+
-#if LUA_VERSION_NUM < 503
-    return
-#endif
     lua_sethook(M_WXLSTATEDATA->m_lua_State, func, mask, count);
+    // lua_sethook returns 1 for lua 5.1 & 5.2
+    // lua_sethook is void in 5.3+
+    return 1; 
```

Comment:
a change is needed because for LUA_VERSION_NUM >= this function wxLuaState::lua_SetHook does not return a value but it should because this function is declared to return int.

You could write 

#if LUA_VERSION_NUM < 503
    return     lua_sethook(M_WXLSTATEDATA->m_lua_State, func, mask, count);
#else
     lua_sethook(M_WXLSTATEDATA->m_lua_State, func, mask, count);
     return 1;
#endif

but this is more verbose and in fact equivalent to the code that I wrote, because lua_sethook always returns 1 for lua 5.1 & 5.2, as I deduced from the lua source code:

LUA_API int lua_sethook (lua_State *L, lua_Hook func, int mask, int count) {
  if (func == NULL || mask == 0) {  /* turn off hooks? */
    mask = 0;
    func = NULL;
  }
  L->hook = func;
  L->basehookcount = count;
  resethookcount(L);
  L->hookmask = cast_byte(mask);
  return 1;
}

or 

LUA_API int lua_sethook (lua_State *L, lua_Hook func, int mask, int count) {
  if (func == NULL || mask == 0) {  /* turn off hooks? */
    mask = 0;
    func = NULL;
  }
  if (isLua(L->ci))
    L->oldpc = L->ci->u.l.savedpc;
  L->hook = func;
  L->basehookcount = count;
  resethookcount(L);
  L->hookmask = cast_byte(mask);
  return 1;
}

Reply:
Another option could be to make the method `wxLuaState::lua_SetHook` void.

Reply:
That's correct, but `lua_sethook` is `void` in Lua 5.3, so I think a proper fix is to restore the original logic and add condition to the return value of `wxLuaState::lua_SetHook`. Do you want me to do this?

Reply:
I'm sorry, but I don't understand the question nor the problem. Do you want to declare `wxLuaState::lua_SetHook` to return type `void` in the header with a preprocessor condition? This at least would solve the compiler warning. But anyway, it seems that the reuturn value is nowhere needed.So do it as you like.

Reply:
I thought again over this question: Perhaps it's best to declare `wxLuaState::lua_SetHook` as void in the header. So noone would get the idea to evaluate the useless return code and this is also the simplest solution.

Reply:
I meant doing something like this:

```
#if LUA_VERSION_NUM < 503
int
#else
void
#endif
wxLuaState::lua_SetHook(lua_Hook func, int mask, int count)
{
     wxCHECK_MSG(Ok(), 0, wxT("Invalid wxLuaState"));
    // lua_sethook is void in 5.3
#if LUA_VERSION_NUM < 503
    return
#endif
     lua_sethook(M_WXLSTATEDATA->m_lua_State, func, mask, count);
```

This should address the warning and match the signature for lua_sethook, right?

Reply:
It's connected to the question what the purpose of the  `wxLuaState` class is: 
- should this class provide a *one-to-one-mapping* to the lua api: than there should be an `#if` in the header (as in your example) und you get several version of `wxLuaState` per proprocessor defines for different lua api versions. But in this case, there  also should be an `#if` for `luaL_register` and the missing `luaL_register` method must be handled outside the `wxLuaState` class for lua5.3.
- or should the class `wxLuaState` be a kind of abstraction layer: than this layer should be the same for different lua versions and than there should be no `#if` for different lua versions. In this case there should be an `luaL_register` (or similar) that works the same for different lua versions and `lua_sethook` could have return type `void` for all lua versions.